Bug: splunk-operator subscription stuck UpgradePending

When running the audit for OpenShift 4.8 the WaitForCsvOnNamespace fails:

```sh
Melvins-MacBook-Pro:opcap mrhillsman$ build/opcap check
I0722 01:08:44.870356   33816 request.go:601] Waited for 1.049241572s due to client-side throttling, not priority and fairness, request: GET:https://api.4-8-audit-install-config.opdev.io:6443/apis/image.openshift.io/v1?timeout=32s
check called
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x380 pc=0x2134739]

goroutine 1 [running]:
opcap/internal/operator.operatorClient.WaitForCsvOnNamespace({{0xc00021fce0?, 0x27b57b0?}}, {0xc001fa21f8, 0x15})
        /Users/mrhillsman/Development/goland/opcap/internal/operator/csv.go:57 +0x2f9
opcap/internal/capability.(*capAudit).OperatorInstall(0xc00164d2c0)
        /Users/mrhillsman/Development/goland/opcap/internal/capability/operator_install.go:42 +0x442
reflect.Value.call({0x2312f00?, 0xc00164d2c0?, 0x1?}, {0x250e7d4, 0x4}, {0x0, 0x0, 0x27c08e8?})
        /usr/local/opt/go/libexec/src/reflect/value.go:556 +0x845
reflect.Value.Call({0x2312f00?, 0xc00164d2c0?, 0xc001f7aed0?}, {0x0, 0x0, 0x0})
        /usr/local/opt/go/libexec/src/reflect/value.go:339 +0xbf
opcap/internal/capability.(*capAuditor).RunAudits(0xc002c7bd50)
        /Users/mrhillsman/Development/goland/opcap/internal/capability/auditor.go:78 +0x1f5
opcap/cmd.glob..func2(0x33e7ac0?, {0x250ecd0?, 0x0?, 0x0?})
        /Users/mrhillsman/Development/goland/opcap/cmd/check.go:62 +0xef
github.com/spf13/cobra.(*Command).execute(0x33e7ac0, {0x3433038, 0x0, 0x0})
        /Users/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x33e7d40)
        /Users/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/mrhillsman/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
opcap/cmd.Execute()
        /Users/mrhillsman/Development/goland/opcap/cmd/root.go:36 +0x25
main.main()
        /Users/mrhillsman/Development/goland/opcap/main.go:10 +0x17
Melvins-MacBook-Pro:opcap mrhillsman$
```

![splunk](https://user-images.githubusercontent.com/3989695/180446053-659dd3f4-76dc-4546-9fdf-0d9b8b043a49.png)

Further investigation needs to be made to determine if we can reasonably handle such an error and continue audit.

--- a/cmd/check.go
+++ b/cmd/check.go
@@ -93,8 +93,7 @@ type CheckCommandFlags struct {
 var checkflags CheckCommandFlags
 
 func init() {
-
-	var defaultAuditPlan = []string{"OperatorInstall", "OperatorCleanUp"}
+	defaultAuditPlan := []string{"OperatorInstall", "OperatorCleanUp"}
 
 	rootCmd.AddCommand(checkCmd)
 	flags := checkCmd.Flags()

--- a/cmd/version.go
+++ b/cmd/version.go
@@ -13,11 +13,13 @@ func init() {
 // Those variables are populated at build time by ldflags.
 // If you're running from a local debugger they will show empty fields.
 
-var Version string
-var GoVersion string
-var BuildTime string
-var GitUser string
-var GitCommit string
+var (
+	Version   string
+	GoVersion string
+	BuildTime string
+	GitUser   string
+	GitCommit string
+)
 
 var versionCmd = &cobra.Command{
 	Use:   "version",

--- a/go.mod
+++ b/go.mod
@@ -96,7 +96,7 @@ require (
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
-	k8s.io/apiextensions-apiserver v0.24.0 // indirect
+	k8s.io/apiextensions-apiserver v0.24.0
 	k8s.io/component-base v0.24.0 // indirect
 	k8s.io/klog/v2 v2.60.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42 // indirect

--- a/internal/capability/audit.go
+++ b/internal/capability/audit.go
@@ -12,7 +12,6 @@ import (
 
 // CapAudit is an implementation of the Audit interface
 type capAudit struct {
-
 	// client has access to all operator methods
 	client operator.Client
 
@@ -51,7 +50,6 @@ type capAudit struct {
 }
 
 func newCapAudit(c operator.Client, subscription operator.SubscriptionData, auditPlan []string) (capAudit, error) {
-
 	ns := strings.Join([]string{"opcap", strings.ReplaceAll(subscription.Package, ".", "-")}, "-")
 	operatorGroupName := strings.Join([]string{subscription.Name, subscription.Channel, "group"}, "-")
 
@@ -74,7 +72,6 @@ func newCapAudit(c operator.Client, subscription operator.SubscriptionData, audi
 }
 
 func newOperatorGroupData(name string, targetNamespaces []string) operator.OperatorGroupData {
-
 	return operator.OperatorGroupData{
 		Name:             name,
 		TargetNamespaces: targetNamespaces,
@@ -82,7 +79,6 @@ func newOperatorGroupData(name string, targetNamespaces []string) operator.Opera
 }
 
 func getTargetNamespaces(s operator.SubscriptionData, namespace string) []string {
-
 	targetNs1 := strings.Join([]string{namespace, "targetns1"}, "-")
 	targetNs2 := strings.Join([]string{namespace, "targetns2"}, "-")
 

--- a/internal/capability/auditor.go
+++ b/internal/capability/auditor.go
@@ -8,7 +8,6 @@ import (
 
 // capAuditor implements Auditor
 type CapAuditor struct {
-
 	// AuditPlan holds the tests that should be run during an audit
 	AuditPlan []string
 
@@ -26,7 +25,6 @@ type CapAuditor struct {
 
 // BuildWorkQueueByCatalog fills in the auditor workqueue with all package information found in a specific catalog
 func (capAuditor *CapAuditor) buildWorkQueueByCatalog() error {
-
 	c, err := operator.NewOpCapClient()
 	if err != nil {
 		// if it doesn't load the client nothing can be done
@@ -71,7 +69,6 @@ func (capAuditor *CapAuditor) RunAudits() error {
 
 	// read workqueue for audits
 	for audit := range capAuditor.WorkQueue {
-
 		// read a particular audit's auditPlan for functions
 		// to be executed against operator
 		for _, function := range audit.auditPlan {
@@ -80,7 +77,6 @@ func (capAuditor *CapAuditor) RunAudits() error {
 			m := reflect.ValueOf(&audit).MethodByName(function)
 			m.Call(nil)
 		}
-
 	}
 	return nil
 }

--- a/internal/capability/operand_install.go
+++ b/internal/capability/operand_install.go
@@ -113,7 +113,7 @@ func (ca *capAudit) OperandInstall() error {
 		ca.operands = append(ca.operands, *unstructuredCR)
 	}
 
-	file, err := os.OpenFile("operand_install_report.json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	file, err := os.OpenFile("operand_install_report.json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
 	if err != nil {
 		file.Close()
 		return err

--- a/internal/capability/operator_cleanup.go
+++ b/internal/capability/operator_cleanup.go
@@ -7,7 +7,6 @@ import (
 )
 
 func (ca *capAudit) OperatorCleanUp() error {
-
 	// delete subscription
 	if err := ca.client.DeleteSubscription(context.Background(), ca.subscription.Name, ca.namespace); err != nil {
 		logger.Debugf("Error while deleting Subscription: %w", err)

--- a/internal/capability/operator_install.go
+++ b/internal/capability/operator_install.go
@@ -47,7 +47,7 @@ func (ca *capAudit) OperatorInstall() error {
 		}
 	}
 
-	file, err := os.OpenFile("operator_install_report.json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	file, err := os.OpenFile("operator_install_report.json", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
 	if err != nil {
 		file.Close()
 		return err

--- a/internal/capability/report.go
+++ b/internal/capability/report.go
@@ -3,14 +3,12 @@ package capability
 import (
 	"fmt"
 	"io"
-
 	"time"
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 func (ca capAudit) OperatorInstallTextReport(w io.Writer) error {
-
 	fmt.Fprint(w, "\n")
 	fmt.Fprint(w, "Operator Install Report:\n")
 	fmt.Fprint(w, "-----------------------------------------\n")
@@ -35,7 +33,6 @@ func (ca capAudit) OperatorInstallTextReport(w io.Writer) error {
 }
 
 func (ca capAudit) OperatorInstallJsonReport(w io.Writer) error {
-
 	if !ca.csvTimeout {
 		fmt.Fprintf(w, "{\"level\":\"info\",\"message\":\""+string(ca.csv.Status.Phase)+"\",\"package\":\""+ca.subscription.Package+
 			"\",\"channel\":\""+ca.subscription.Channel+"\",\"installmode\":\""+string(ca.subscription.InstallModeType)+"\"}\n")
@@ -48,7 +45,6 @@ func (ca capAudit) OperatorInstallJsonReport(w io.Writer) error {
 }
 
 func (ca capAudit) OperandTextReport(w io.Writer) error {
-
 	for _, cr := range ca.customResources {
 		operand := &unstructured.Unstructured{Object: cr}
 
@@ -72,16 +68,13 @@ func (ca capAudit) OperandTextReport(w io.Writer) error {
 }
 
 func (ca capAudit) OperandInstallJsonReport(w io.Writer) error {
-
 	for _, cr := range ca.customResources {
 		operand := &unstructured.Unstructured{Object: cr}
 
 		if len(ca.operands) > 0 {
-
 			fmt.Fprintf(w, "{\"package\":\""+ca.subscription.Package+"\", \"Operand Kind\": \""+operand.GetKind()+"\", \"Operand Name\": \""+operand.GetName()+
 				"\",\"message\":\""+"created"+"\"}\n")
 		} else {
-
 			fmt.Fprintf(w, "{\"package\":\""+ca.subscription.Package+"\", \"Operand Kind\": \""+operand.GetKind()+"\", \"Operand Name\": \""+operand.GetName()+
 				"\",\"message\":\""+"failed"+"\"}\n")
 		}

--- a/internal/operator/csv.go
+++ b/internal/operator/csv.go
@@ -11,9 +11,8 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 )
 
-//Gets completed CSVs, Succeeded or Failed, with timeout on delay duration
+// Gets completed CSVs, Succeeded or Failed, with timeout on delay duration
 func (c operatorClient) GetCompletedCsvWithTimeout(namespace string, delay time.Duration) (operatorv1alpha1.ClusterServiceVersion, error) {
-
 	// csv will catch CSVs from watch events
 	csv := &operatorv1alpha1.ClusterServiceVersion{}
 	var ok bool
@@ -38,10 +37,9 @@ func (c operatorClient) GetCompletedCsvWithTimeout(namespace string, delay time.
 	defer ticker.Stop()
 
 	for range ticker.C {
-
 		select {
 
-		//case catches CSV events
+		// case catches CSV events
 		case event := <-eventChan:
 			csv, ok = event.Object.(*operatorv1alpha1.ClusterServiceVersion)
 			// fail on wrong objects
@@ -70,7 +68,6 @@ func (c operatorClient) GetCompletedCsvWithTimeout(namespace string, delay time.
 
 // waits for CSV on namespace and gets a watcher for CSV events
 func (c operatorClient) csvWatcher(namespace string) (watch.Interface, error) {
-
 	ctx := context.Background()
 	var watcher watch.Interface
 
@@ -92,7 +89,6 @@ func (c operatorClient) csvWatcher(namespace string) (watch.Interface, error) {
 
 			return true, nil
 		})
-
 	if err != nil {
 		logger.Error("Failed to create csv.")
 		return nil, err

--- a/internal/operator/subscription.go
+++ b/internal/operator/subscription.go
@@ -113,7 +113,7 @@ func (c operatorClient) ListPackageManifests(ctx context.Context, list *pkgserve
 }
 
 func filterPackageManifests(manifests []pkgserverv1.PackageManifest, catalogSource string, filter []string) []pkgserverv1.PackageManifest {
-	var result = []pkgserverv1.PackageManifest{}
+	result := []pkgserverv1.PackageManifest{}
 
 	if len(filter) == 0 && catalogSource == "" {
 		return manifests

--- a/internal/operator/subscription_test.go
+++ b/internal/operator/subscription_test.go
@@ -55,7 +55,7 @@ var manifests = []pkgserverv1.PackageManifest{
 }
 
 func TestFilterPackageManifests(t *testing.T) {
-	var cases = []struct {
+	cases := []struct {
 		catalog string
 		filters []string
 		expect  int

